### DISCUSSION TITLE
It's possible to rename tenant

### DISCUSSION THREAD

**Discussion by lenglet-k** [grafana/mimir#6220] [Help and support] [answered]
Hello,

We have old tenant in this format: "my_tenant:namespace".

And today we have enable tenant_federation option and the ":" caracter is forbidenn. So we don't want lost old tenant i want know if it's possible to rename old tenants with the new tenant name in this format: `my_tenant_namespace`

Thanks you

**Comment by dimitarvdimitrov** [comment, marked as answer]
It's not an explicit feature but it should be possible. The safest way to do this is to
* stop writing metrics to the old tenant for 13h
	* in the meantime you can just switch to writing to the new tenant name; alternatively write to both for the time you need metrics available
* stop querying the old tenant and query only the new tenant
* in the object store change the prefix with the tenant name to the new tenant; something like `mv s3://bucket/my_tenant:namespace/01203010120BK/chunks/0000001 s3://bucket/my_tenant/01203010120BK/chunks/0000001` for each object
	* skip copying the `bucket-index.json.gz` file; otherwise you'd see some query unavailability under the enw tenant
* you'd also have to change the tenant name in overrides in the runtime config or if it's used it in any other Mimir configuration 


I can't guarantee this will work flawlessly, but it should do the job. None of the files in object store contain the name of the tenant, apart from in the prefix, so moving them is safe. The bucket-index is only a snapshot of the block locations, so moving it over just means the snapshot is a stale one.

**Reply by lenglet-k** [reply]
Hi, i have done that. it's ok. Thanks you. I have rename s3 bucket from old tenant to new tenant.
